Remove FullscreenManager::FullscreenPromise

achristensen07 opened the pull request:
#### 4df703fdc7cc8ca2f8371b7b260f66259def169b
<pre>
Remove FullscreenManager::FullscreenPromise
<a href="https://bugs.webkit.org/show_bug.cgi?id=287898">https://bugs.webkit.org/show_bug.cgi?id=287898</a>
<a href="https://rdar.apple.com/145092075">rdar://145092075</a>

Reviewed by Eric Carlson.

The concept of storing the promise on the FullscreenManager was added in 257668@main
to make sure it was called in all cases.  Now that I&apos;ve gone through the entire flow
of entering and exiting fullscreen and used CompletionHandlers through the whole
flow, we can just pass along a CompletionHandler from begin to end of enter and exit.

* Source/WebCore/dom/FullscreenManager.cpp:
(WebCore::FullscreenManager::requestFullscreenForElement):
(WebCore::FullscreenManager::cancelFullscreen):
(WebCore::FullscreenManager::exitFullscreen):
(WebCore::FullscreenManager::willEnterFullscreen):
(WebCore::FullscreenManager::didExitFullscreen):
(WebCore::FullscreenManager::clear):
(WebCore::FullscreenManager::FullscreenPromise::~FullscreenPromise): Deleted.
(WebCore::FullscreenManager::FullscreenPromise::operator=): Deleted.
(WebCore::FullscreenManager::FullscreenPromise::clear): Deleted.
(WebCore::FullscreenManager::FullscreenPromise::resolve): Deleted.
(WebCore::FullscreenManager::FullscreenPromise::rejectOrResolve): Deleted.
(WebCore::FullscreenManager::FullscreenPromise::reject): Deleted.
(WebCore::FullscreenManager::FullscreenPromise::operator bool const): Deleted.
* Source/WebCore/dom/FullscreenManager.h:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::exitFullScreenForElement):
* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
(WebKit::WebFullScreenManager::exitFullScreenForElement):
(WebKit::WebFullScreenManager::willExitFullScreen):
(WebKit::WebFullScreenManager::didExitFullScreen):
* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::exitFullScreenForElement):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::dispatchDidStartProvisionalLoad):
* Source/WebKitLegacy/mac/WebCoreSupport/WebChromeClient.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebChromeClient.mm:
(WebChromeClient::exitFullScreenForElement):
* Source/WebKitLegacy/mac/WebCoreSupport/WebKitFullScreenListener.mm:
(-[WebKitFullScreenListener webkitDidExitFullScreen]):
* Source/WebKitLegacy/mac/WebView/WebFullScreenController.h:
* Source/WebKitLegacy/mac/WebView/WebFullScreenController.mm:
(-[WebFullScreenController exitFullScreen:]):
(-[WebFullScreenController finishedExitFullScreenAnimation:]):
(-[WebFullScreenController close]):
(-[WebFullScreenController exitFullScreen]): Deleted.
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _exitFullScreenForElement:completionHandler:]):
(-[WebView _exitFullScreenForElement:]): Deleted.
* Source/WebKitLegacy/mac/WebView/WebViewInternal.h:

Canonical link: <a href="https://commits.webkit.org/290619@main">https://commits.webkit.org/290619@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4b97aacc1fe970f5209e5f1734cdee4142a64527

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/90614 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/10146 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/45554 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/95626 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/41396 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/92668 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/10544 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/18465 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/69733 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/27294 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/93615 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/8048 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/82180 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/50082 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/7794 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/36558 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/40526 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/78128 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/37622 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/97455 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/17806 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/13096 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/78757 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/18064 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/78003 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/77954 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/22393 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/21018 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/11128 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/14261 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/17816 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/23155 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/17555 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/21011 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/19339 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/dom/FullscreenManager.cpp:
```diff
@@ -563,21 +557,14 @@ bool FullscreenManager::willExitFullscreen()
-bool FullscreenManager::didExitFullscreen()
+void FullscreenManager::didExitFullscreen(CompletionHandler<void(ExceptionOr<void>)>&& completionHandler)
 {
     auto fullscreenElement = fullscreenOrPendingElement();
```
Can you move this to line 576 where this is first used?

Something like this:
```
    if (auto fullscreenElement = fullscreenOrPendingElement())
        fullscreenElement->didStopBeingFullscreenElement();
```

nt1m reviewed (commented):
WebCore parts + flow LGTM, though I'm not familiar enough with the WebKit bits to review them.

achristensen07 committed:
Remove FullscreenManager::FullscreenPromise
https://bugs.webkit.org/show_bug.cgi?id=287898
rdar://145092075

Reviewed by Eric Carlson.

The concept of storing the promise on the FullscreenManager was added in 257668@main
to make sure it was called in all cases.  Now that I've gone through the entire flow
of entering and exiting fullscreen and used CompletionHandlers through the whole
flow, we can just pass along a CompletionHandler from begin to end of enter and exit.

* Source/WebCore/dom/FullscreenManager.cpp:
(WebCore::FullscreenManager::requestFullscreenForElement):
(WebCore::FullscreenManager::cancelFullscreen):
(WebCore::FullscreenManager::exitFullscreen):
(WebCore::FullscreenManager::willEnterFullscreen):
(WebCore::FullscreenManager::didExitFullscreen):
(WebCore::FullscreenManager::clear):
(WebCore::FullscreenManager::FullscreenPromise::~FullscreenPromise): Deleted.
(WebCore::FullscreenManager::FullscreenPromise::operator=): Deleted.
(WebCore::FullscreenManager::FullscreenPromise::clear): Deleted.
(WebCore::FullscreenManager::FullscreenPromise::resolve): Deleted.
(WebCore::FullscreenManager::FullscreenPromise::rejectOrResolve): Deleted.
(WebCore::FullscreenManager::FullscreenPromise::reject): Deleted.
(WebCore::FullscreenManager::FullscreenPromise::operator bool const): Deleted.
* Source/WebCore/dom/FullscreenManager.h:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::exitFullScreenForElement):
* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.cpp:
(WebKit::WebFullScreenManager::exitFullScreenForElement):
(WebKit::WebFullScreenManager::willExitFullScreen):
(WebKit::WebFullScreenManager::didExitFullScreen):
* Source/WebKit/WebProcess/FullScreen/WebFullScreenManager.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::exitFullScreenForElement):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::dispatchDidStartProvisionalLoad):
* Source/WebKitLegacy/mac/WebCoreSupport/WebChromeClient.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebChromeClient.mm:
(WebChromeClient::exitFullScreenForElement):
* Source/WebKitLegacy/mac/WebCoreSupport/WebKitFullScreenListener.mm:
(-[WebKitFullScreenListener webkitDidExitFullScreen]):
* Source/WebKitLegacy/mac/WebView/WebFullScreenController.h:
* Source/WebKitLegacy/mac/WebView/WebFullScreenController.mm:
(-[WebFullScreenController exitFullScreen:]):
(-[WebFullScreenController finishedExitFullScreenAnimation:]):
(-[WebFullScreenController close]):
(-[WebFullScreenController exitFullScreen]): Deleted.
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView _exitFullScreenForElement:completionHandler:]):
(-[WebView _exitFullScreenForElement:]): Deleted.
* Source/WebKitLegacy/mac/WebView/WebViewInternal.h:

Canonical link: https://commits.webkit.org/290619@main
